cedar-policy-symcc/src/symcc/extension_types/ipaddr.rs

```diff
@@ -22,15 +22,27 @@
 
 use std::sync::LazyLock;
 
+use miette::Diagnostic;
 use num_bigint::BigUint;
+use thiserror::Error;
 
 use crate::symcc::{
-    bitvec::BitVec,
-    result::Error,
+    bitvec::{BitVec, BitVecError},
     type_abbrevs::{nat, Fin, Nat, Width},
 };
 
-type Result<T> = std::result::Result<T, Error>;
+/// Errors in [`IPNet`] operations.
+#[derive(Debug, Diagnostic, Error)]
+pub enum IPError {
+    /// Errors in [`BitVec`] operations.
+    #[error("bit-vector error when manipulating ip addresses")]
```

Comment:
We should include the text from the underlying `BitVecError` here.

```suggestion
    #[error("bit-vector error when manipulating ip addresses: {0}")]
```

Reply:
Ah that's where I wasn't particularly sure about.

Since we derive the `thiserror::Error` trait, it should implement the [`Error::source`](https://doc.rust-lang.org/std/error/trait.Error.html#method.source), which should point to the underlying error (BitVecError in this case). And then at the top-level, when we do `format!("{:?}", miette::Report::from(err))`, we should get the entire chain of errors, making the `: {0}` part redundant?

But if the user just do `format!("{:?}", err)`, it may actually just print the top-level error. So I'm not sure what's the best practice here

Reply:
ooh I see